jpos/src/main/java/org/jpos/emv/IssuerApplicationData.java

```diff
@@ -70,7 +70,7 @@ public IssuerApplicationData(String hexIAD) {
             //Therefore, the length of the Issuer Application Data is 18 bytes and for 
             // M/Chip Advance it may be 18, 20, 26, or 28.
             unpackMCHIP(iad);
-        } else if (len == 32 && iad.startsWith("0F") && iad.startsWith("0F", 32)) {
+        } else if (len == 32 && (iad.startsWith("0F") || iad.startsWith("0f")) && (iad.startsWith("0F", 32) || iad.startsWith("0f", 32))) {
```

Comment:
A few stylistic suggestions:
* you may want to do a `toUpperCase()` (or lower...) after the `trim()` in line 60, to simplify the expression here.
* Renaming the variable `len` to `byteLen` will clarify you're not talking about the string but the bytes.
* writing `16*2` instead of `32` will make it clear as well (it confused us for a minute), and the compiler is smart enough to make it a constant 😉

Reply:
>you may want to do a toUpperCase() (or lower...) after the trim() in line 60 

this was my initial thought, but it ended up breaking some tests so  this is minimal way of fixing it.

- I was thinking of making the padding method  to take the Block size  n  as an argument as well, so it would be n*2  instead and it should work with 16 byte AES blocks as well. but I need to test it first.